primefaces/src/main/java/org/primefaces/util/BeanPropertyComparator.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2009-2023 PrimeTek Informatics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.primefaces.util;
+
+import java.text.Collator;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Locale;
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.el.ValueExpression;
+import javax.faces.FacesException;
+import javax.faces.context.FacesContext;
+
+import org.primefaces.component.api.UITable;
+import org.primefaces.context.PrimeApplicationContext;
+import org.primefaces.model.SortMeta;
+
+public class BeanPropertyComparator implements Comparator<Object> {
```

Comment:
currently the name sounds like a independent reusable class, but its heavily bound to UITable.
Feels like we should move+rename

Reply:
I used same name as before (PF 8 or 10). But indeed not a great name... I renamed into `SortMetaComparator` (which 100% correct) but if you have a better name i'm all ears